modules/internal/ChapelHashtable.chpl

```diff
@@ -495,7 +501,8 @@ module ChapelHashtable {
     }
 
     proc maybeShrinkAfterRemove() {
-      if (tableNumFullSlots*8 < tableSize) {
+      if (tableSize > startingSize &&
+          tableNumFullSlots/tableSize:real < resizeThreshold/4) {
```

Comment:
is the `4` special here? Did it change from `8`? Could you add a comment and/or lift the `4` into a `const`?

Reply:
I'll add a comment for it. The 4 corresponds to our map being a power of 2 table size, whereas if it is 1/4 of the resizeThreshold, that means it will be 1/2 of the resizeThreshold once we divide the table size in half, which seems like a good time to resize, but it really is just an arbitrary decision (that was made in a conversation with Brad [here](https://github.com/Cray/chapel-private/issues/2626))

Reply:
Looks good, nice explanation